### PR TITLE
ref-responses

### PR DESCRIPTION
**Pull request by annashamray**
Issue - https://github.com/VNG-Realisatie/gemma-zaken/issues/1582

### CODE REVIEW

File: vng_api_common/schema.py [hunk 1]
```diff
@@ -135,3 +174,16 @@ def get(self, request, version="", *args, **kwargs):
             server["url"] = request.build_absolute_uri(server_path)
 
         return Response(data=schema, headers={"X-OAS-Version": schema["openapi"]})
+
+
+class YamlView(View):
```
Comment (joeribekker) [inline vng_api_common/schema.py:179]: I don't see a need to serve it, the unresolved version of the YAML file is just for people who want to download the spec, which can be done via Github. Let's remove the view.

File: bin/generate_schema [hunk 2]
```diff
@@ -20,6 +20,7 @@ src/manage.py generate_swagger \
 echo "Converting Swagger to OpenAPI 3.0..."
 npm run convert
 patch_content_types
+use_external_components
```
Comment (joeribekker) [inline bin/generate_schema:23]: Can we generate 2 schema files?

One called `openapi.yaml` which include everything (with internal refs)
One called `openapi_unresolved.yaml` which includes via external refs from `common.yaml`

Comment (annashamray) [reply bin/generate_schema:23]: `openapi_unresolved.yaml` is generated with refs to `common.yaml` url

File: vng_api_common/schema.py (hunk 1 above)
Comment (annashamray) [reply vng_api_common/schema.py:179]: the view is removed

**Comment by sergei-maertens** [comment]
@annashamray this PR has been in limbo for a long time now, so I discussed it with Joeri.

Can you confirm that using this exact implementation does not break the schema generation in gemma-zaakregistratiecomponent or Open Zaak if you run `generate_schema` without a `common.yaml`?

If that is indeed backwards compatible without issues, we'll merge and release this.

**Comment by stevenbal** [comment]
> @annashamray this PR has been in limbo for a long time now, so I discussed it with Joeri.
> 
> Can you confirm that using this exact implementation does not break the schema generation in gemma-zaakregistratiecomponent or Open Zaak if you run `generate_schema` without a `common.yaml`?
> 
> If that is indeed backwards compatible without issues, we'll merge and release this.

@sergei-maertens do you mean whether schema generation still works, even if `common.yaml` cannot be retrieved from the specified `COMMON_SPEC` url? Because that is currently not the case, so I could make some changes to fix that

**Comment by sergei-maertens** [comment]
@annashamray no, rather that if there is no `COMMON_SPEC` used (yet), that the schema-generation does not result in changes in the schema whatsoever.

**Comment by stevenbal** [comment]
> @annashamray no, rather that if there is no `COMMON_SPEC` used (yet), that the schema-generation does not result in changes in the schema whatsoever.

It does result in some changes, namely that the response bodies are declared once in `components` and referred to (which wasn't the case before)
Screenshot from 2020-10-16 10-06-26

**Comment by sergei-maertens** [comment]
Thanks for testing this, I think this is an acceptable change as the semantics stay the same.

**Comment by annashamray** [comment]
@sergei-maertens @stevenbal sorry, I've missed your comments.
The idea of this PR is to make a generated schema more compact, to do this two things were introduced:
1. All response objects with statuses >= 400 are moved into `components/`, and then reused as references to the components, It helped to created response objects for fault responses once, and not for every endpoint

2. If you have `common.yaml` file with specified components, the components from the original file would be replaced with them.

So if you don't have `common.yaml` file you'll just have your response objects rearranged in the schema file and have your schema a little more compact

**Pull request merged**